### PR TITLE
feat: support lists on recently viewed screen

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [ONYX-679]

### Description

This PR adds view as list on the recently viewed screen.

**What I suggest to pay attention to**
- I found a better way to support list/grid without need to make an additional request ;) I achieved that by tweaking the number of columns with Flashlist Masonry component. As a next step, I will use the same logic in the NWFY screen and save us one additional request 💪 
  - This will also make it easier to support the same behaviour in other screens

https://github.com/artsy/eigen/assets/11945712/dcaf862e-1b0a-4f45-a6bb-3a4ee49b8496

https://github.com/artsy/eigen/assets/11945712/0fd5228c-763f-4774-9331-8215594af96d

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [X] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- support lists on recently viewed screen - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-679]: https://artsyproduct.atlassian.net/browse/ONYX-679?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (support lists on recently viewed screen - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 78a16855a5a655c7ec5db06df7a8559029ce44eb
</p>

**Review by dariakoko — DISMISSED** [review]
Great work ⭐

File: src/app/Scenes/RecentlyViewed/RecentlyViewed.tests.tsx [hunk 1]
```diff
@@ -1,51 +1,23 @@
...
-    mockEnvironment.mock.resolveMostRecentOperation((operation) =>
-      MockPayloadGenerator.generate(operation, {
-        Query: () => mockResponse,
-      })
-    )
+    await flushPromiseQueue()
 
-    expect(tree.findByText("Recently Viewed")).toBeTruthy()
-    expect(tree.findByText("Sunflower Seeds Exhibition")).toBeTruthy()
+    expect(screen.getByText("Sunflower Seeds Exhibition")).toBeTruthy()
```
Comment (gkartalis) [inline src/app/Scenes/RecentlyViewed/RecentlyViewed.tests.tsx:18]: **suggestion(non-blocking)**: could you change the `toBeTruthy` with `toBeOnTheScreen()` ?

File: src/app/Scenes/RecentlyViewed/RecentlyViewed.tests.tsx [hunk 2]
```diff
@@ -1,51 +1,23 @@
...
-
-  it("renders RecentlyViewed", () => {
-    const tree = renderWithWrappers(<TestRenderer />)
-
-    mockEnvironment.mock.resolveMostRecentOperation((operation) =>
-      MockPayloadGenerator.generate(operation, {
-        Query: () => mockResponse,
-      })
-    )
+    await flushPromiseQueue()
```
Comment (gkartalis) [inline src/app/Scenes/RecentlyViewed/RecentlyViewed.tests.tsx:16]: **suggestion(non-blocking)**: we could get away with not using flushPromiseQueue here by adding a testID on the skeleton and do something like this instead:
```ts
await waitForElementToBeRemoved(() => screen.getByTestId("RecentlyViewedScreenPlaceholder"))
```

**Review by gkartalis — APPROVED** [review]
awesome work! Thanks for adding the migration as well, left two non blocking comments for the tests

File: src/app/Scenes/RecentlyViewed/RecentlyViewed.tests.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/app/Scenes/RecentlyViewed/RecentlyViewed.tests.tsx:18]: good idea!

**Comment by MounirDhahri** [comment]
Will merge this to unblock the work on the other screens, glad to address comments separately

**Pull request merged**